### ISSUE TITLE
Associated Legendre polynomials are requested 

### ISSUE THREAD

**Issue by feima0011** [JuliaMath/SpecialFunctions.jl#124]
Associated Legendre polynomials are requested 
https://en.wikipedia.org/wiki/Associated_Legendre_polynomials

**Comment by ludoro** [comment]
I would love to try and implement these! Working on it.

**Comment by simonbyrne** [comment]
@feima0011 Can you expand on how you're using them? i.e. are you evaluating the same polynomial multiple times? If so, is it on a grid?

**Comment by feima0011** [comment]
> @feima0011 Can you expand on how you're using them? i.e. are you evaluating the same polynomial multiple times? If so, is it on a grid?

I am working on the acoustic wave equations. I need the Associated Legendre polynomials
to calculate the spherical harmonics.  I have to calculate the value of the Associated Legendre polynomials multiple times, and they are on  a grid.

**Comment by simonbyrne** [comment]
Do you have a link to code where they are used?

**Comment by MikaelSlevinsky** [comment]
Have you tried using the spherical harmonica transforms in FastTransforms.jl? There is an example on usage here https://github.com/MikaelSlevinsky/FastTransforms.jl/blob/master/examples/sphere.jl

**Comment by feima0011** [comment]
Hi Richard Mikael Slevinsky :

I have heard that package before.
I think a standalone code to compute the Legendre polynomial
is better than the code in that package.

fei

**Comment by arturgower** [comment]
I agree with Fei. Associated Legendre polynomials, and [spherical harmonics](https://en.wikipedia.org/wiki/Spherical_harmonics), are core to so many branches of physics. They involve many connections to other special functions and complicated identities. It makes sense to have them embedded in SpecialFunctions.jl.

**Comment by PaulXiCao** [comment]
We could start at Legendre Polynomials, then Associated Legendre Polynomials and then Spherical Harmonics?

Concerning Legendre Polynomials
We should implement a standard method using the recurrenc relation which is alright for small orders.
For higher orders I found a [paper](https://biblio.ugent.be/publication/3108575/file/3108618.pdf) which describes an algorithm to compute Legendre Polynomials in constant time by precomputation. It is quite involved. I am not sure if we should include that in SpecialFunctions. Maybe a standalone package is better?

**Comment by PaulXiCao** [comment]
The associated Legendre polynomials (functions of first kind) have been implemented by a simple recursion formula in pr #175 (one of the later commits).

**Comment by LeoGalPras** [comment]
Should this be merged with issues #175 and #177 ???
